### PR TITLE
Added Berdy MAP solver

### PR DESCRIPTION
**Pull request by francesco-romano**
In the process I also implemented a ColumnMajor version of SparseMatrix.
This is needed as almost all the Eigen sparse solvers (especially the Cholesky) support only column major ordering.

Tests (locally) pass, even if we do not have a lot of tests for SparseMatrix and Berdy.
Pending:
Windows compilation

### CODE REVIEW

**Comment by francesco-romano** [comment]
Berdy solver compilation is disabled for Eigen < 3.3 (as it is in Ubuntu 14.04(

**Comment by francesco-romano** [comment]
Thanks to @drdanz swig does not segfault anymore.
Still to test the bindings in Matlab

**Comment by francesco-romano** [comment]
@traversaro I am missing the bindings for #include `BerdySparseMAPSolver`.
That class can be compiled only with Eigen >= 3.3
Should I add it conditionally also on `iDynTree.i` ?

**Comment by traversaro** [comment]
This is tricky: having a conditional dependency on the bindings means that that dependency is compulsory when compiling octave/matlab (as the idea is that users do not regenerate matlab/octave bindings). If it is necessary, I would be in favor of just depending on Eigen >= 3.3 , and let users of old distro get a new Eigen using the superbuild). However, I don't know the status of https://github.com/ocra-recipes/ocra-recipes/issues/26 , and I am a bit afraid of making the latest version of iDynTree and orca-recepies mutually exclusive. 
@rlober @jeljaik Any news?

File: src/estimation/CMakeLists.txt [hunk 1]
```diff
@@ -4,6 +4,27 @@
...
+set(CMAKE_CXX_EXTENSIONS OFF)
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED 11)
+if(NOT CMAKE_MINIMUM_REQUIRED_VERSION VERSION_LESS 3.1)
+  message(AUTHOR_WARNING "CMAKE_MINIMUM_REQUIRED_VERSION is now ${CMAKE_MINIMUM_REQUIRED_VERSION}. This check can be removed.")
+endif()
+if(${CMAKE_VERSION} VERSION_LESS 3.1)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${CXX11_FLAGS}")
+endif()
+
```
Comment (traversaro) [inline src/estimation/CMakeLists.txt]: Can you use the `idyntree_enable_cxx11` macro?

**Comment by jeljaik** [comment]
Hello @traversaro !!!
Yes, we have internally migrated to KDL to avoid using Eigen LGSM but we're still missing a few tests. Within the next week this should be done indeed.

**Comment by traversaro** [comment]
@jeljaik  Great!

File: src/estimation/CMakeLists.txt (hunk 1 above)
Comment (francesco-romano) [inline src/estimation/CMakeLists.txt]: Done

**Comment by francesco-romano** [comment]
So, if I am correct, we should wait for @jeljaik and @rlober before merging? @traversaro

**Comment by francesco-romano** [comment]
I should also add a copy constructor between different ordering

**Comment by traversaro** [comment]
If @jeljaik is confident that ocra will be compatible with Eigen 3.3 in the near future, in the meanwhile we can make the Eigen 3.3 dependency compulsory in the `devel` branch and make sure that we do a release (i.e. merge of `devel` in `master`) only when they will have completed the KDL migration. 

Actually before requiring Eigen 3.3 in the `devel` branch it would be better to be sure that everything works fine in Debian Jessie, that have Eigen 3.2 as a system package. A possible clean solution is to require Eigen 3.3 in the codyco-superbuild level, meaning that if Eigen 3.3 is not found in the system it will be downloaded as any superbuild project (see https://github.com/robotology/codyco-superbuild/blob/master/cmake/BuildEigen3.cmake). However, this will need to be thoroughly tested as it is possible that the superbuild packages will still find the system Eigen3 instead of the superbuild one. In the short term, we can just require Eigen 3.3 and make sure that everything on the iCubGenova02 `pc104` system works fine (at this is the main system in IIT for which we care that iDynTree compiles fine).

**Comment by traversaro** [comment]
`jessie-backports` contains Eigen3 (see https://packages.debian.org/jessie-backports/libeigen3-dev). That is a great clean solution, so I think that at least in the `devel` branch we can simply require Eigen 3.3 .

**Comment by francesco-romano** [comment]
With my "small" commit I broke something.
Travis is blocked for some YCM error (I restarted the job).
AppVeyor is failing some tests.

**Comment by traversaro** [comment]
Good old memcheck has some nice results: 
~~~
24/109 Test  #24: memcheck_UnitTestSparseMatrix .............................***Failed    1.07 sec
==31770== Memcheck, a memory error detector
==31770== Copyright (C) 2002-2013, and GNU GPL'd, by Julian Seward et al.
==31770== Using Valgrind-3.10.1 and LibVEX; rerun with -h for copyright info
==31770== Command: /home/travis/build/robotology/idyntree/build/bin/SparseMatrixUnitTest
==31770== 
Testing RowMajor ordering
------------------------------------
testCreateMatrixFromAccessorOperator
Matrix:
0 3 0 0 0 
22 0 0 0 17 
7 5 0 1 0 
0 0 0 0 0 
0 0 14 0 8 

------------------------------------
------------------------------------
testCreateMatrixFromTriplets
Matrix:
-6 3 0 0 0 
22 0 0 0 17 
7 5 0 1 0 
0 0 0 0 0 
0 0 14 0 8 

------------------------------------
------------------------------------
testCreateMatrixFromDuplicateTriplets
Matrix:
-1 7 0 0 0 
22 0 0 0 12 
7 5 0 1 0 
0 0 0 0 0 
0 0 14 0 8 

------------------------------------
------------------------------------
testMatrixIterator
4(0,1)
23(1,0)
18(1,4)
8(2,0)
6(2,1)
2(2,3)
15(4,2)
9(4,4)
------------------------------------
------------------------------------
testZeroingMatrix

------------------------------------
------------------------------------
testClassesTraits
is_trivially_destructible:
Triplet: true
Triplets: false
------------------------------------
Testing ColumnMajor ordering
------------------------------------
testCreateMatrixFromAccessorOperator
Matrix:
0 3 0 0 0 
22 0 0 0 17 
7 5 0 1 0 
0 0 0 0 0 
0 0 14 0 8 

------------------------------------
------------------------------------
testCreateMatrixFromTriplets
Matrix:
-6 3 0 0 0 
22 0 0 0 17 
7 5 0 1 0 
0 0 0 0 0 
0 0 14 0 8 

------------------------------------
------------------------------------
testCreateMatrixFromDuplicateTriplets
Matrix:
-1 7 0 0 0 
22 0 0 0 12 
7 5 0 1 0 
0 0 0 0 0 
0 0 14 0 8 

------------------------------------
Testing RowMajor-ColumnMajor conversions
------------------------------------
testRowColumnMajorConversion
Initial (row Major) matrix
0 3 0 0 0 
22 0 0 0 17 
7 5 0 1 0 
0 0 0 0 0 
0 0 14 0 8 


==31770== Conditional jump or move depends on uninitialised value(s)
==31770==    at 0x5225494: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::resize(unsigned int, unsigned int, iDynTree::VectorDynSize const&) (SparseMatrix.cpp:390)
==31770==    by 0x52274EC: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::resize(unsigned int, unsigned int) (SparseMatrix.cpp:89)
==31770==    by 0x5225A59: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::SparseMatrix<(iDynTree::MatrixStorageOrdering)1>(iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)1> const&) (SparseMatrix.cpp:476)
==31770==    by 0x4049C1: testRowColumnMajorConversion(iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>) (SparseMatrixUnitTest.cpp:189)
==31770==    by 0x404F63: main (SparseMatrixUnitTest.cpp:237)
==31770== 
==31770== Conditional jump or move depends on uninitialised value(s)
==31770==    at 0x5224141: iDynTree::VectorDynSize::reserve(unsigned int) (VectorDynSize.cpp:185)
==31770==    by 0x5226B83: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::initializeMatrix(unsigned int, double const*, unsigned int) (SparseMatrix.cpp:45)
==31770==    by 0x52254EC: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::resize(unsigned int, unsigned int, iDynTree::VectorDynSize const&) (SparseMatrix.cpp:396)
==31770==    by 0x52274EC: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::resize(unsigned int, unsigned int) (SparseMatrix.cpp:89)
==31770==    by 0x5225A59: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::SparseMatrix<(iDynTree::MatrixStorageOrdering)1>(iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)1> const&) (SparseMatrix.cpp:476)
==31770==    by 0x4049C1: testRowColumnMajorConversion(iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>) (SparseMatrixUnitTest.cpp:189)
==31770==    by 0x404F63: main (SparseMatrixUnitTest.cpp:237)
==31770== 
==31770== Conditional jump or move depends on uninitialised value(s)
==31770==    at 0x5223FBE: iDynTree::VectorDynSize::changeCapacityAndCopyData(unsigned int) (VectorDynSize.cpp:159)
==31770==    by 0x5224157: iDynTree::VectorDynSize::reserve(unsigned int) (VectorDynSize.cpp:190)
==31770==    by 0x5226B83: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::initializeMatrix(unsigned int, double const*, unsigned int) (SparseMatrix.cpp:45)
==31770==    by 0x52254EC: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::resize(unsigned int, unsigned int, iDynTree::VectorDynSize const&) (SparseMatrix.cpp:396)
==31770==    by 0x52274EC: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::resize(unsigned int, unsigned int) (SparseMatrix.cpp:89)
==31770==    by 0x5225A59: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::SparseMatrix<(iDynTree::MatrixStorageOrdering)1>(iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)1> const&) (SparseMatrix.cpp:476)
==31770==    by 0x4049C1: testRowColumnMajorConversion(iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>) (SparseMatrixUnitTest.cpp:189)
==31770==    by 0x404F63: main (SparseMatrixUnitTest.cpp:237)
==31770== 
==31770== Conditional jump or move depends on uninitialised value(s)
==31770==    at 0x4C2B7AD: operator new[](unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==31770==    by 0x5224042: iDynTree::VectorDynSize::changeCapacityAndCopyData(unsigned int) (VectorDynSize.cpp:167)
==31770==    by 0x5224157: iDynTree::VectorDynSize::reserve(unsigned int) (VectorDynSize.cpp:190)
==31770==    by 0x5226B83: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::initializeMatrix(unsigned int, double const*, unsigned int) (SparseMatrix.cpp:45)
==31770==    by 0x52254EC: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::resize(unsigned int, unsigned int, iDynTree::VectorDynSize const&) (SparseMatrix.cpp:396)
==31770==    by 0x52274EC: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::resize(unsigned int, unsigned int) (SparseMatrix.cpp:89)
==31770==    by 0x5225A59: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::SparseMatrix<(iDynTree::MatrixStorageOrdering)1>(iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)1> const&) (SparseMatrix.cpp:476)
==31770==    by 0x4049C1: testRowColumnMajorConversion(iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>) (SparseMatrixUnitTest.cpp:189)
==31770==    by 0x404F63: main (SparseMatrixUnitTest.cpp:237)
==31770== 
==31770== Conditional jump or move depends on uninitialised value(s)
==31770==    at 0x5227E2F: std::vector<int, std::allocator<int> >::reserve(unsigned long) (vector.tcc:68)
==31770==    by 0x5226B9C: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::initializeMatrix(unsigned int, double const*, unsigned int) (SparseMatrix.cpp:48)
==31770==    by 0x52254EC: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::resize(unsigned int, unsigned int, iDynTree::VectorDynSize const&) (SparseMatrix.cpp:396)
==31770==    by 0x52274EC: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::resize(unsigned int, unsigned int) (SparseMatrix.cpp:89)
==31770==    by 0x5225A59: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::SparseMatrix<(iDynTree::MatrixStorageOrdering)1>(iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)1> const&) (SparseMatrix.cpp:476)
==31770==    by 0x4049C1: testRowColumnMajorConversion(iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>) (SparseMatrixUnitTest.cpp:189)
==31770==    by 0x404F63: main (SparseMatrixUnitTest.cpp:237)
==31770== 
==31770== Conditional jump or move depends on uninitialised value(s)
==31770==    at 0x5227E4E: std::vector<int, std::allocator<int> >::reserve(unsigned long) (vector.tcc:70)
==31770==    by 0x5226B9C: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::initializeMatrix(unsigned int, double const*, unsigned int) (SparseMatrix.cpp:48)
==31770==    by 0x52254EC: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::resize(unsigned int, unsigned int, iDynTree::VectorDynSize const&) (SparseMatrix.cpp:396)
==31770==    by 0x52274EC: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::resize(unsigned int, unsigned int) (SparseMatrix.cpp:89)
==31770==    by 0x5225A59: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::SparseMatrix<(iDynTree::MatrixStorageOrdering)1>(iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)1> const&) (SparseMatrix.cpp:476)
==31770==    by 0x4049C1: testRowColumnMajorConversion(iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>) (SparseMatrixUnitTest.cpp:189)
==31770==    by 0x404F63: main (SparseMatrixUnitTest.cpp:237)
==31770== 
==31770== Conditional jump or move depends on uninitialised value(s)
==31770==    at 0x409C60: std::_Vector_base<int, std::allocator<int> >::_M_allocate(unsigned long) (stl_vector.h:168)
==31770==    by 0x522B067: int* std::vector<int, std::allocator<int> >::_M_allocate_and_copy<std::move_iterator<int*> >(unsigned long, std::move_iterator<int*>, std::move_iterator<int*>) (stl_vector.h:1138)
==31770==    by 0x5227E9E: std::vector<int, std::allocator<int> >::reserve(unsigned long) (vector.tcc:73)
==31770==    by 0x5226B9C: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::initializeMatrix(unsigned int, double const*, unsigned int) (SparseMatrix.cpp:48)
==31770==    by 0x52254EC: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::resize(unsigned int, unsigned int, iDynTree::VectorDynSize const&) (SparseMatrix.cpp:396)
==31770==    by 0x52274EC: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::resize(unsigned int, unsigned int) (SparseMatrix.cpp:89)
==31770==    by 0x5225A59: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::SparseMatrix<(iDynTree::MatrixStorageOrdering)1>(iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)1> const&) (SparseMatrix.cpp:476)
==31770==    by 0x4049C1: testRowColumnMajorConversion(iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>) (SparseMatrixUnitTest.cpp:189)
==31770==    by 0x404F63: main (SparseMatrixUnitTest.cpp:237)
==31770== 
==31770== Conditional jump or move depends on uninitialised value(s)
==31770==    at 0x409CDC: __gnu_cxx::new_allocator<int>::allocate(unsigned long, void const*) (new_allocator.h:101)
==31770==    by 0x409C7F: std::_Vector_base<int, std::allocator<int> >::_M_allocate(unsigned long) (stl_vector.h:168)
==31770==    by 0x522B067: int* std::vector<int, std::allocator<int> >::_M_allocate_and_copy<std::move_iterator<int*> >(unsigned long, std::move_iterator<int*>, std::move_iterator<int*>) (stl_vector.h:1138)
==31770==    by 0x5227E9E: std::vector<int, std::allocator<int> >::reserve(unsigned long) (vector.tcc:73)
==31770==    by 0x5226B9C: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::initializeMatrix(unsigned int, double const*, unsigned int) (SparseMatrix.cpp:48)
==31770==    by 0x52254EC: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::resize(unsigned int, unsigned int, iDynTree::VectorDynSize const&) (SparseMatrix.cpp:396)
==31770==    by 0x52274EC: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::resize(unsigned int, unsigned int) (SparseMatrix.cpp:89)
==31770==    by 0x5225A59: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::SparseMatrix<(iDynTree::MatrixStorageOrdering)1>(iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)1> const&) (SparseMatrix.cpp:476)
==31770==    by 0x4049C1: testRowColumnMajorConversion(iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>) (SparseMatrixUnitTest.cpp:189)
==31770==    by 0x404F63: main (SparseMatrixUnitTest.cpp:237)
==31770== 
==31770== Conditional jump or move depends on uninitialised value(s)
==31770==    at 0x4C2B08D: operator new(unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==31770==    by 0x409CF6: __gnu_cxx::new_allocator<int>::allocate(unsigned long, void const*) (new_allocator.h:104)
==31770==    by 0x409C7F: std::_Vector_base<int, std::allocator<int> >::_M_allocate(unsigned long) (stl_vector.h:168)
==31770==    by 0x522B067: int* std::vector<int, std::allocator<int> >::_M_allocate_and_copy<std::move_iterator<int*> >(unsigned long, std::move_iterator<int*>, std::move_iterator<int*>) (stl_vector.h:1138)
==31770==    by 0x5227E9E: std::vector<int, std::allocator<int> >::reserve(unsigned long) (vector.tcc:73)
==31770==    by 0x5226B9C: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::initializeMatrix(unsigned int, double const*, unsigned int) (SparseMatrix.cpp:48)
==31770==    by 0x52254EC: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::resize(unsigned int, unsigned int, iDynTree::VectorDynSize const&) (SparseMatrix.cpp:396)
==31770==    by 0x52274EC: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::resize(unsigned int, unsigned int) (SparseMatrix.cpp:89)
==31770==    by 0x5225A59: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::SparseMatrix<(iDynTree::MatrixStorageOrdering)1>(iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)1> const&) (SparseMatrix.cpp:476)
==31770==    by 0x4049C1: testRowColumnMajorConversion(iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>) (SparseMatrixUnitTest.cpp:189)
==31770==    by 0x404F63: main (SparseMatrixUnitTest.cpp:237)
==31770== 
==31770== Conditional jump or move depends on uninitialised value(s)
==31770==    at 0x522706D: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::reserve(unsigned int) (SparseMatrix.cpp:96)
==31770==    by 0x522554D: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::setFromTriplets(iDynTree::Triplets&) (SparseMatrix.cpp:406)
==31770==    by 0x5225AB3: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::SparseMatrix<(iDynTree::MatrixStorageOrdering)1>(iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)1> const&) (SparseMatrix.cpp:481)
==31770==    by 0x4049C1: testRowColumnMajorConversion(iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>) (SparseMatrixUnitTest.cpp:189)
==31770==    by 0x404F63: main (SparseMatrixUnitTest.cpp:237)
==31770== 
==31770== Conditional jump or move depends on uninitialised value(s)
==31770==    at 0x522410D: iDynTree::VectorDynSize::reserve(unsigned int) (VectorDynSize.cpp:183)
==31770==    by 0x5227083: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::reserve(unsigned int) (SparseMatrix.cpp:98)
==31770==    by 0x522554D: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::setFromTriplets(iDynTree::Triplets&) (SparseMatrix.cpp:406)
==31770==    by 0x5225AB3: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::SparseMatrix<(iDynTree::MatrixStorageOrdering)1>(iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)1> const&) (SparseMatrix.cpp:481)
==31770==    by 0x4049C1: testRowColumnMajorConversion(iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>) (SparseMatrixUnitTest.cpp:189)
==31770==    by 0x404F63: main (SparseMatrixUnitTest.cpp:237)
==31770== 
==31770== Conditional jump or move depends on uninitialised value(s)
==31770==    at 0x5224141: iDynTree::VectorDynSize::reserve(unsigned int) (VectorDynSize.cpp:185)
==31770==    by 0x5227083: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::reserve(unsigned int) (SparseMatrix.cpp:98)
==31770==    by 0x522554D: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::setFromTriplets(iDynTree::Triplets&) (SparseMatrix.cpp:406)
==31770==    by 0x5225AB3: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::SparseMatrix<(iDynTree::MatrixStorageOrdering)1>(iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)1> const&) (SparseMatrix.cpp:481)
==31770==    by 0x4049C1: testRowColumnMajorConversion(iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>) (SparseMatrixUnitTest.cpp:189)
==31770==    by 0x404F63: main (SparseMatrixUnitTest.cpp:237)
==31770== 
==31770== Conditional jump or move depends on uninitialised value(s)
==31770==    at 0x5227E4E: std::vector<int, std::allocator<int> >::reserve(unsigned long) (vector.tcc:70)
==31770==    by 0x5227096: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::reserve(unsigned int) (SparseMatrix.cpp:99)
==31770==    by 0x522554D: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::setFromTriplets(iDynTree::Triplets&) (SparseMatrix.cpp:406)
==31770==    by 0x5225AB3: iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>::SparseMatrix<(iDynTree::MatrixStorageOrdering)1>(iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)1> const&) (SparseMatrix.cpp:481)
==31770==    by 0x4049C1: testRowColumnMajorConversion(iDynTree::SparseMatrix<(iDynTree::MatrixStorageOrdering)0>) (SparseMatrixUnitTest.cpp:189)
==31770==    by 0x404F63: main (SparseMatrixUnitTest.cpp:237)
==31770== 
------------------------------------


------------------------------------
testRowColumnMajorConversion
Initial (row Major) matrix
0 0 5 0 0 
0 0 0 0 8.8 
0 0 1.8 0 0 
0 0 0 0 0 
0 0 0 0 0 


------------------------------------
==31770== 
==31770== HEAP SUMMARY:
==31770==     in use at exit: 0 bytes in 0 blocks
==31770==   total heap usage: 295 allocs, 295 frees, 31,814 bytes allocated
==31770== 
==31770== All heap blocks were freed -- no leaks are possible
==31770== 
==31770== For counts of detected and suppressed errors, rerun with: -v
==31770== Use --track-origins=yes to see where uninitialised values come from
==31770== ERROR SUMMARY: 26 errors from 13 contexts (suppressed: 0 from 0)
~~~

**Comment by francesco-romano** [comment]
Let's see the results of the new tests on Travis, as they were never run given the old Eigen version

**Comment by francesco-romano** [comment]
Forgot to add the bindings for the Berdy solver 😢

**Comment by francesco-romano** [comment]
This time I should have fixed everything. Let's wait for CI

**Pull request merged**